core/build.gradle

```diff
@@ -10,18 +10,18 @@ plugins {
 version = '0.17-SNAPSHOT'
 
 dependencies {
-    api 'org.bouncycastle:bcprov-jdk15to18:1.70'
-    api 'com.google.guava:guava:31.0.1-android'
+    api 'org.bouncycastle:bcprov-jdk15to18:1.71'
```

Comment:
If you revert this to 1.70 the rest of the PR works.

Reply:
If the missing BouncyCastle turns out to be more than a mistake, I will remove the BC update from this PR.